packages/anchor_token/src/querier.rs

```diff
@@ -54,31 +47,23 @@ pub fn query_token_balance<S: Storage, A: Api, Q: Querier>(
     Ok(balance.into())
 }
 
-pub fn query_supply<S: Storage, A: Api, Q: Querier>(
-    deps: &Extern<S, A, Q>,
-    contract_addr: &HumanAddr,
-) -> StdResult<Uint256> {
+pub fn query_supply(deps: Deps, contract_addr: Addr) -> StdResult<Uint256> {
     // load price form the oracle
     let res: Binary = deps.querier.query(&QueryRequest::Wasm(WasmQuery::Raw {
```

Comment:
If we need to support both old and new, we need to query twice 
length_prefixed and not prefixed. 

or just using Smart query is better